src/main/java/picard/sam/SamAlignmentMerger.java

```diff
@@ -158,6 +141,70 @@ public void mergeAlignment(final File referenceFasta) {
         }
     }
 
+    @Override
+    protected SAMSequenceDictionary getDictionaryForMergedBam() {
+        SAMSequenceDictionary finalDict = null;
+        if (alignedSamFile != null && !alignedSamFile.isEmpty()) {
+            for (final File alignedFile : alignedSamFile) {
```

Comment:
Couldn't you just `map()` to dictionaries, then `foreach()` to compare each to the first dictionary, in ~2 lines?